protobuf-maven-plugin/src/main/java/io/github/ascopes/protobufmavenplugin/generate/SourceCodeGenerator.java

```diff
@@ -104,6 +117,11 @@ public boolean generate(GenerationRequest request) throws ResolutionException, I
       }
     }
 
+    if (resolvedPlugins.isEmpty() && hasPlugins(request) && pluginsOptional(request)) {
```

Comment:
There shouldn't be any errors later on, since if you look above we do the same thing when `isFailOnMissingSources()` is set

Reply:
just checking this again... does this cover not providing language outputs?

E.g. suppose I make some protobuf sources in src/main/protobuf, and then set javaEnabled to false. If I provide optional plugins that don't resolve, then it may get to invoking protoc without any language output and without any plugins, at which point protoc will fail.